src/components/TextField/TextField.css.ts

```diff
@@ -0,0 +1,128 @@
+import { style } from '@vanilla-extract/css';
+import { recipe } from '@vanilla-extract/recipes';
+import { palette } from '@/styles/color';
+import { variants } from '../common/typography/Typography.css';
+
+export const inputTypeRecipe = recipe({
+  variants: {
+    status: {
+      default: 'default',
+      active: 'active',
+      error: 'error',
+      loading: 'loading',
+      success: 'success'
+    }
+  },
+  compoundVariants: [
+    {
+      variants: {},
+      style: {
+        vars: {
+          '--status-color': palette.gray300
+        }
+      }
+    },
+    {
+      variants: { status: 'active' },
+      style: {
+        vars: {
+          '--status-color': palette.gray900
+        }
+      }
+    },
+    {
+      variants: { status: 'error' },
+      style: {
+        vars: {
+          '--status-color': palette.error
+        }
+      }
+    },
+    {
+      variants: { status: 'success' },
+      style: {
+        vars: {
+          '--status-color': palette.primary300
+        }
+      }
+    }
+  ]
+});
+
+export const wrapper = style({
+  display: 'flex',
+  flexDirection: 'column',
+  position: 'relative'
+});
+
+export const label = style({
+  color: palette.gray600,
+  marginBottom: '6px'
+});
+
+export const inputContainer = style({
+  position: 'relative',
+  maxWidth: '400px',
+  display: 'flex',
+  alignItems: 'center',
+  boxSizing: 'border-box',
+  marginBottom: '8px',
+  '::placeholder': {
+    color: palette.gray200
+  }
+});
+
+export const input = recipe({
+  variants: {
+    size: {
+      big: variants.h3,
+      small: variants.body2
+    }
+  },
+
+  compoundVariants: [
+    {
+      variants: {},
+      style: {
+        position: 'relative',
+        marginBottom: '6px',
+        width: '100%'
+      }
+    }
+  ]
+});
+
+export const underbar = style({
+  position: 'absolute',
+  width: `100%`,
+  height: '1px',
+  borderBottom: '1px solid black',
+  bottom: '0px',
+  borderColor: 'var(--status-color)'
+});
+
+export const icon = recipe({
+  base: {
+    visibility: 'hidden'
+  },
+  variants: {
+    status: {
+      on: {
+        visibility: 'visible'
+      },
+      off: {
+        visibility: 'hidden'
+      }
+    }
```

Comment:
```suggestion
    visible: {
      true: {
        visibility: 'visible'
      },
      false: {
        visibility: 'hidden'
      }
    }
```
on이거나 off인 양자 택일 값이면 boolean을 이용하는 게 더 깔끔할 것 같아요
이렇게 바꾸면 boolean 값으로 쉽게 variant를 설정할 수 있습니다
```typescript
  icon({ visible: true })
```

Reply:
recipe variants에 과연 boolean 값으로 접근할수 있을까? 라는 생각만 갖고 이런 방식으로 작성했었는데,, 수정하겠습니다!!